### DISCUSSION TITLE
How to change the file name

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#2787] [Q&A] [answered]
how can I change the filename for uploaded files or is there any way to disable suffixing 10 numbers at the end of the file name after uploading them to Pocketbase.

**Comment by ganigeorgiev** [comment]
The name of the uploaded files cannot be changed (at least not without hacky workarounds). If you want to visualize the original filename to the client you can store it as another text field during the upload. There are plans in the future to add support for new endpoint returning the file attributes (format, size and maybe original name), but for now is a low priority.

**Comment by nikhil-varma-mudunuri** [comment]
I am planning to use pocketBase as an HLS server, for which I need to specify filenames in the .m3u8 file (like file01.m4a, file02.m4a ...) but when I upload the files in pocketBase, files names are modified which is the problem I am facing. 
my solution is : 
first, upload all files except the .m3u8 file and now fetch the filename from pocketBase and modify those names in the .m3u8 file and now upload the .m3u8 file to pocket base

are there any other solutions?  thanks.

**Reply by ganigeorgiev** [reply, marked as answer]
The only solution I can think of at the moment is to attach to `OnRecordBeforeCreateRequest`/`OnRecordBeforeUpdateRequest` hooks and modify the [`e.UploadedFiles`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.5/core#RecordCreateEvent) slice.

The problem is that you'll also need to replace the filenames references in the record data. I'll consider in the next v0.17/v0.18 to allow changing the uploaded files by just replacing the `e.UploadedFiles[].Name` field, but for now you can try:

```go
package main

import (
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	// preserve the original filenames on create
	app.OnRecordBeforeCreateRequest("yourCollection").Add(func(e *core.RecordCreateEvent) error {
		oldNames := map[string]map[string]string{}

		for field, files := range e.UploadedFiles {
			if len(files) == 0 {
				continue
			}

			oldNames[field] = make(map[string]string, len(files))
			for _, f := range files {
				oldNames[field][f.Name] = f.OriginalName

				f.Name = f.OriginalName
			}
		}

		// update the record file fields with the new names
		for field, filenames := range oldNames {
			files := e.Record.GetStringSlice(field)

			for i, old := range files {
				if new, ok := filenames[old]; ok {
					files[i] = new
				}
			}

			e.Record.Set(field, files)
		}

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```
